client/src/components/TypeList.vue

```diff
@@ -19,6 +19,43 @@ export default Vue.extend({
       required: true,
     },
   },
+  data() {
+    return {
+      showPicker: false,
+      selectedColor: '',
+      selectedType: '',
+      editingType: '',
+      editingColor: '',
+    };
+  },
+  methods: {
+    clickEdit(type: string) {
+      this.selectedType = type;
+      this.editingType = this.selectedType;
+      this.showPicker = true;
+      this.selectedColor = this.typeColorMapper(type);
+      this.editingColor = this.selectedColor;
+    },
+    acceptChanges() {
+      this.showPicker = false;
+      if (this.editingType !== this.selectedType) {
+        this.$emit('update-type-name', { currentType: this.selectedType, newType: this.editingType });
+      }
+      if (this.editingColor !== this.selectedColor) {
+        this.$emit('update-type-color', { type: this.editingType, color: this.editingColor });
+      }
+      this.colorRefresh();
+    },
+    /**
+     * Causes the color to refresh for annotations and events by toggling an item
+     */
+    colorRefresh() {
+      const val = this.checkedTypes.value.pop();
+      if (val) {
+        this.checkedTypes.value.push(val); // Causes a color refresh
```

Comment:
I don't think this is the right way to go about triggering an update.  

`typeColorMapper` should be a computed property dependent on the state of `dataset.meta`.  This only works for UI that rendered based on the state of `checkedTypes`, or in other words, the filtered list.  If we had a list of all tracks, or something else not derived from that, this wouldn't work and would probably confuse whoever was debugging it.

If you're busy I can take a swing at this.

Reply:
Yeah I was going to start yesterday moving it into a computed property but got sidetracked with some other stuff.  Most of today I'll probably working on other stuff.  I didn't like this way of doing it either and was trying to find a better way when I asked the other day.  Reminder that doing it this way means that the Layers.vue will probably need to watch the `typeColorMap` property to update the annotations in real time.